tests/dir.rs

```diff
@@ -1070,8 +1070,84 @@ fn it_copy_with_progress_exist_overwrite_and_skip_exist() {
 
 }
 
+#[test]
+fn it_copy_mirror_work() {
```

Comment:
One test not enough for this operation. I see 3 situation( command for examples will `cp -r d1 d2`)

**First**

d2 not exist 

``` tree.
.
└── d1
   ├── f1
   └── sub
      └── f2
```
*result*
``` tree.
.
└── d1
   ├── f1
   └── sub
      └── f2
└── d2
   ├── f1
   └── sub
      └── f2

```


**Second**

d2 exist but not contains d1 folder

``` tree.
.
└── d1
   ├── f1
   └── sub
      └── f2
└── d2
      └── fzz
```
*result*
``` tree.
.
└── d1
   ├── f1
   └── sub
      └── f2
└── d2
    └── d1
       ├── f1
       └── sub
          └── f2
    └── fzz

```

**Third**

d2 exist and contains some files of d1 folder

``` tree.
.
└── d1
   ├── f1
   └── sub
      └── f2
└── d2
    └── d1
       ├── f1
       └── sub
          └── f2

```
*result*
``` tree.
.
└── d1
   ├── f1
   └── sub
      └── f2
└── d2
    └── d1
       ├── f1   // will be overwrite if will be set flag overwrite
       └── sub
          └── f2 // will be overwrite if will be set flag overwrite
    └── fzz

```

Reply:
Hi, webdesus. I think the situation 2&3 can be simply done by checking d2 folder and and it's sub folder and then determine whether to copy to the sub folder. 
Maybe I just implement the cp command with a rename feature as situation 1. :blush:

Reply:
I think better will be write test before:wink:

Reply:
Three test cases add!